src/layer/LayerGroup.js

```diff
@@ -50,6 +52,26 @@ export const LayerGroup = Layer.extend({
 		return this;
 	},
 
+	_preventLayerGroupRecursion(layer) {
+		if (layer instanceof LayerGroup && layer.hasDescendant(this)) {
+			throw new Error('Cannot add a LayerGroup as a child if it is already included, directly or nested.');
+		}
+	},
+
+	// @method hasDescendant(layer: Layer): Boolean
+	// Returns `true` if the given layer is already a child of this instance, directly or nested.
+	hasDescendant(layer) {
+		let hasDescendant = false;
+
+		this.eachLayer((childLayer) => {
+			if (childLayer === layer || (childLayer instanceof LayerGroup && childLayer.hasDescendant(layer))) {
+				hasDescendant = true;
+			}
+		});

```

Comment:
Changing this to a for-loop would make it possible to break it if a descendant has been found.